src/system.c

```diff
@@ -259,8 +259,10 @@ void register_out_columns(
         if (columns[i].inout_kind == EcsOut) {
             if (!system_data->on_demand) {
                 system_data->on_demand = ecs_os_malloc(sizeof(ecs_on_demand_out_t));
-                system_data->on_demand->system = system;
-                system_data->on_demand->count = 0;
+				if (system_data->on_demand != NULL) {
```

Comment:
Indentation. Also, while it is good to check for the return of malloc, the pattern in the code has been to assert if it returns `NULL` with a
```c
ecs_assert(system_data->on_demand != NULL, ECS_OUT_OF_MEMORY, NULL);
```

The reasoning behind this is that otherwise the code is left in an undefined state (the system has not been properly created) from which it cannot recover.